source/includes/indexes/single-field.scala

```diff
@@ -0,0 +1,39 @@
+// start-single-index-imports
+import org.mongodb.scala._
+import org.mongodb.scala.model.Indexes
+import org.mongodb.scala.model.IndexOptions._
+import org.mongodb.scala.model.Projections._
+import org.mongodb.scala.model.Sorts._
+import org.mongodb.scala.model.Filters._
+// end-single-index-imports
+
+
+object SingleFieldIndex {
+
+    def main(args: Array[String]): Unit = {
+
+    // Replace the placeholder with your Atlas connection string
+    val connectionString = "<connection string>";
+
+    // Create a new client and connect to the server
+    val mongoClient = MongoClient(connectionString)
+    val database = mongoClient.getDatabase("sample_training")
+    val collection = database.getCollection("companies")
+
+    // start-index-single
+    collection.createIndex(Indexes.ascending("title")).printResults()
```

Comment:
The `printResults` method is not defined by the Scala driver.  When I paste this file into IntelliJ it tells me:

```
Cannot resolve symbol printResults
```

and the compiler tells me:

```
[Error] /Users/jeff.yemin/git/mongo-java-driver/driver-scala/src/integration/scala/tour/SingleFieldIndex.scala:25:56: value printResults is not a member of org.mongodb.scala.SingleObservable[String]
[Error] /Users/jeff.yemin/git/mongo-java-driver/driver-scala/src/integration/scala/tour/SingleFieldIndex.scala:38:30: value printResults is not a member of org.mongodb.scala.ListIndexesObservable[org.mongodb.scala.Document]
two errors found
```

How are you verifying that this code is valid?

Reply:
I am testing the code locally. The `printResults` method can be found in the quickstart as a helpful method for printing query results: https://deploy-preview-70--docs-scala.netlify.app/get-started/connect-to-mongodb/#add-helper-methods. I can include a note that `Helpers` is being used if this is the correct way to print the query results. 

It also seems that without using `printResults` when the index is created, the index doesn't show up when the `listIndexes` method is called with `printResults`

Can you explain what the `printResults` method is doing? Is there a better way to print the results of the query? Thanks! @jyemin

Reply:
Sure, let's have a look.  Here's `printResults`:

```
def printResults(initial: String = ""): Unit = {
    if (initial.length > 0) print(initial)
    results().foreach(res => println(converter(res)))
}
```

It doesn't do much of interest itself besides printing, but it does call the `results` method:

```
def results(): Seq[C] = Await.result(observable.toFuture(), Duration(10, TimeUnit.SECONDS))
```

This method does two interesting things:

1. It calls `toFuture` on the `Observable` 
2. It waits for 10 seconds for the result of the Future to become available

The first is key to answering your question, I think.  An `Observable` returned by the Scala driver is "cold" to start.  So if you just do:

```
collection.createIndex(Indexes.ascending("title")).
```

nothing happens.  It's only when the application takes some action to "observe the Observable` that an operation is kicked off. Calling `toFuture` on it is one of those observations, but there are many others (e.g. `subscribe`).

The Java Reactive Streams driver is fundamentally the same, by the way.